components/script/dom/xrreferencespaceevent.rs

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use dom_struct::dom_struct;
+use js::rust::HandleObject;
+use servo_atoms::Atom;
+
+use crate::dom::bindings::codegen::Bindings::EventBinding::Event_Binding::EventMethods;
+use crate::dom::bindings::codegen::Bindings::XRReferenceSpaceEventBinding::{
+    XRReferenceSpaceEventInit, XRReferenceSpaceEventMethods,
+};
+use crate::dom::bindings::error::Fallible;
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::{reflect_dom_object_with_proto, DomObject};
+use crate::dom::bindings::root::{Dom, DomRoot};
+use crate::dom::bindings::str::DOMString;
+use crate::dom::event::Event;
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::window::Window;
+use crate::dom::xrreferencespace::XRReferenceSpace;
+use crate::dom::xrrigidtransform::XRRigidTransform;
+use crate::script_runtime::CanGc;
+
+#[dom_struct]
+pub struct XRReferenceSpaceEvent {
+    event: Event,
+    space: Dom<XRReferenceSpace>,
+    transform: Option<DomRoot<XRRigidTransform>>,
+}
+
+impl XRReferenceSpaceEvent {
+    #[allow(crown::unrooted_must_root)]
+    fn new_inherited(
+        space: &XRReferenceSpace,
+        transform: &Option<DomRoot<XRRigidTransform>>,
+    ) -> XRReferenceSpaceEvent {
+        XRReferenceSpaceEvent {
+            event: Event::new_inherited(),
+            space: Dom::from_ref(space),
+            transform: transform.clone(),
+        }
+    }
+
+    pub fn new(
+        global: &GlobalScope,
+        type_: Atom,
+        bubbles: bool,
+        cancelable: bool,
+        space: &XRReferenceSpace,
+        transform: &Option<DomRoot<XRRigidTransform>>,
+    ) -> DomRoot<XRReferenceSpaceEvent> {
+        Self::new_with_proto(
+            global,
+            None,
+            type_,
+            bubbles,
+            cancelable,
+            space,
+            transform,
+            CanGc::note(),
+        )
+    }
+
+    fn new_with_proto(
+        global: &GlobalScope,
+        proto: Option<HandleObject>,
+        type_: Atom,
+        bubbles: bool,
+        cancelable: bool,
+        space: &XRReferenceSpace,
+        transform: &Option<DomRoot<XRRigidTransform>>,
```

Comment:
Let's use `Option<&XrReferenceSpace>` here instead, and update other the other constructor methods to match.